tests/integration/devicetemplate/suite_test.go

```diff
@@ -0,0 +1,169 @@
+package devicetemplate
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/cnrancher/edge-api-server/pkg/apis/edgeapi.cattle.io/v1alpha1"
+	"github.com/cnrancher/edge-api-server/tests/integration/cluster"
+	"github.com/cnrancher/edge-api-server/tests/integration/framework"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+type DeviceTemplateTestSuite struct {
+	suite.Suite
+	k8s *framework.K8sCli
+
+	exampleRevisionYAML       []byte
+	exampleDeviceTemplateYAML []byte
+
+	revisionGVR       schema.GroupVersionResource
+	deviceTemplateGVR schema.GroupVersionResource
+
+	deviceTemplate *v1alpha1.DeviceTemplate
+	revision       *v1alpha1.DeviceTemplateRevision
+	cancel         context.CancelFunc
+}
+
+func (s *DeviceTemplateTestSuite) SetupSuite() {
+	s.T().Log("SetupSuite")
+	var err error
+
+	err = cluster.StepK3dCluster()
+	s.NoError(err, "SetupTest StepK3dCluster error")
+	if err != nil {
+		cluster.CleanK3dCluster()
```

Comment:
Because the StepK3dCluster step is necessary to succeed in the entire test, and to terminate the test when it fails, but calling FailNow does not go through the TearDownSuite process, you need to additionally try to clear the K3dCluster before terminating the test to prevent accidentally causing the cluster to not be released after the test is terminated.